packages/serverpod_shared/lib/src/config.dart

```diff
@@ -386,6 +402,61 @@ class RedisConfig {
   }
 }
 
+/// Configuration for session logging.
+class SessionLogConfig {
+  /// True if persistent logging (e.g., to Redis) should be enabled.
+  final bool persistentEnabled;
+
+  /// True if console logging should be enabled.
+  final bool consoleEnabled;
+
+  /// Creates a new [SessionLogConfig].
+  SessionLogConfig({
+    required this.persistentEnabled,
+    required this.consoleEnabled,
+  });
+
+  factory SessionLogConfig._fromJson(
+    Map sessionLogConfigJson,
+    String name, {
+    required bool databaseEnabled,
+  }) {
+    _validateJsonConfig(
+      {
+        ServerpodEnv.sessionPersistentLogEnabled.configKey: bool,
+        ServerpodEnv.sessionConsoleLogEnabled.configKey: bool,
+      },
+      sessionLogConfigJson,
+      name,
+    );
+
+    var isSessionPersistentLogEnabled = sessionLogConfigJson[
+            ServerpodEnv.sessionPersistentLogEnabled.configKey] ??
+        false;
+
+    if (isSessionPersistentLogEnabled && !databaseEnabled) {
+      stdout.writeln(
+        'Warning: The `persistentEnabled` setting was enabled in the configuration, but this project was created without database support. '
+        'Persistent logging is only available when the database is enabled, so the value will be overridden and disabled.',
+      );
```

Comment:
Suggestion: when looking at this again, I think we should throw a `StateError` here since it's an invalid state. We should also add a test for it.

Reply:
Done. Additionally, the validation logic has now been moved to a dedicated `_validate` method, which you can view here:

- https://github.com/serverpod/serverpod/blob/b1e86124ce4a4122d9e099f33c124bd42a640035/packages/serverpod_shared/lib/src/config.dart#L447-L456

This method is called after initialization here:

- https://github.com/serverpod/serverpod/blob/b1e86124ce4a4122d9e099f33c124bd42a640035/packages/serverpod_shared/lib/src/config.dart#L60-L62

The reason for this change is to ensure that the validation is enforced regardless of how the `SessionLogConfig` is initialized. Initially, the validation was embedded in the `_fromJson` factory method, but if someone directly instantiated `SessionLogConfig` outside of this factory, the validation would be skipped.

By moving the validation to a separate `_validate` method, it can be called every time the configuration is initialized, ensuring that the check for persistent logging (i.e., it can only be enabled if the database is available) is always performed, no matter how or when the config is created. This guarantees that the configuration remains consistent and avoids any invalid states during initialization.